test/lib-test/jdk/test/lib/format/ArrayDiffTest.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package jdk.test.lib.format;
+
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertEquals;
+
+/*
+ * @test
+ * @summary Check ArrayDiff formatting
+ * @library /test/lib
+ * @run testng jdk.test.lib.format.ArrayDiffTest
+ */
+public class ArrayDiffTest {
```

Comment:
Might be it makes sense to check null values. Also, reversed checks. For example there is first = [1, ..], second = []. But no check for  first = [], second = [1, ..] for some cases,

Reply:
Fixed in the 39992fcba97. That ended in finding a minor issue, actually ([ArrayDiff.java:180](https://github.com/openjdk/jdk/pull/112/commits/39992fcba979a2f6bf7ea73aff92461b13220ee2#diff-374ead0149950fce10be8d9d37cca320L180)). Sometimes reversing can be valuable, thanks!